test/jasmine/tests/modebar_test.js

```diff
@@ -1231,6 +1231,36 @@ describe('ModeBar', function() {
                     buttonClosest.click();
                     expect(gd._fullLayout._cartesianSpikesEnabled).toBe('off');
                 });
+
+                it('should work after clicking on "autoScale2d"', function() {
+                    var buttonAutoScale = selectButton(modeBar, 'autoScale2d');
+                    expect(gd._fullLayout._cartesianSpikesEnabled).toBe('off');
```

Comment:
Currently, clicking on the "autoScale2d" modebar button on a graph that has `_cartesianSpikesEnabled: 'off'` on first draw, results in 

```js
gd._fullLayout._cartesianSpikesEnabled // => 'on'
```

which is obviously wrong.